pylint/testutils/_primer/comparator.py

```diff
@@ -0,0 +1,41 @@
+# Licensed under the GPL: https://www.gnu.org/licenses/old-licenses/gpl-2.0.html
+# For details: https://github.com/PyCQA/pylint/blob/main/LICENSE
+# Copyright (c) https://github.com/PyCQA/pylint/blob/main/CONTRIBUTORS.txt
+
+from __future__ import annotations
+
+import json
+from collections.abc import Generator
+from pathlib import Path
+
+from pylint.testutils._primer.primer_command import Messages, PackageMessages
```

Comment:
Could you do a follow up PR that moves some of these to `typing.py` file?

Reply:
I'm starting to think we should use the existing class for this, I'll give it a try.

Reply:
Resolving as we merged #7000